### PR TITLE
chore: keep the indentation style homogeneous accross IDEs

### PR DESCRIPTION
**Pull request by nulltoken**
Helps maintaining the code style across IDEs

cf. https://editorconfig.org/

**Checklist**

- [ ] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

While working on #635 I noticed that my IDE (VSCode) insisted on trying to format files with four spaces.

This file combined with the VSCode extension (https://editorconfig.org/#download) helps keeping the code style tamed.

### CODE REVIEW

**Comment by philsturgeon** [comment]
We have a special ts lint ruleset, so if you have tslint enabled it should keep things the same, right? Editor config is good, are we added it as a backup for those contributors who are missing tslint ?

**Comment by nulltoken** [comment]
>We have a special ts lint ruleset, so if you have tslint enabled it should keep things the same, right?

I have tslint enabled, and to my best knowledge, none of my files have been staged with an offending indentation. However the husky hook only happens at pre-commit time (behind the back of the developer).

This file (along with the plugin, for the IDEs that don't natively support it) would tidy the file when it's being saved, under the eyes of the committer.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+root = true
+
+[*]
```
Comment (P0lip) [inline .editorconfig]: We seem to be using 4 spaces for certain JSON files, so perhaps let's account for that.

Comment (nulltoken) [reply .editorconfig]: >We seem to be using 4 spaces for certain JSON files, so perhaps let's account for that.

@P0lip I can think of three options:
 - Include in the PR normalized versions (to two spaces wide indentation) of those files (under `/src/meta`, for instance) 
  - If those indentations have to be preserved (to ease diffing comparison with an external source of truth for instance), add another `.editorconfig` file under this hierarchy specifying that `.json` files here and below are expected to be indented with 4 spaces
 - Update the root `.editorconfig` file to replace the `*` with something more restrictive which would explicitly not consider json (eg. `[*.{ts,js,yaml}]`)

Comment (philsturgeon) [reply .editorconfig]: Let's go with `[*.{ts,js,yaml}]` so we can avoid doing anything funky with Markdown files (like trimming intentional trailing whitespace).

Comment (P0lip) [reply .editorconfig]: Yeah.
That said, we can still have charset set to utf-8 for all file types.

Re YAML - I'd stick to indentation only, as trailing spaces and newlines might be intentional in some cases.

Comment (nulltoken) [reply .editorconfig]: @philsturgeon @P0lip Latest proposal should encompass all those requirements above

**Comment by P0lip** [comment]
I'll go ahead and merge it. If Phil has any objections, we can address them in the next PR.

**Pull request merged**